### ISSUE TITLE
MeanAveragePrecision (mAP)

### ISSUE THREAD

**Issue by jl749** [jl749/YOLOv3#7]
## little bit of history
https://learnopencv.com/mean-average-precision-map-object-detection-model-evaluation-metric/

## pycocotool (torchmetric is also based on pycocotool)
https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L378

mAP = torch.mean(precision_curve)
https://github.com/cocodataset/cocoapi/blob/8c9bcc3cf640524c4c20a9c40e89cb6a2f2fa0e9/PythonAPI/pycocotools/cocoeval.py#L439

## yolov5 (based on https://github.com/rafaelpadilla/Object-Detection-Metrics)
https://github.com/rafaelpadilla/Object-Detection-Metrics/blob/e3f29579afef10e8057bda1beb6154a3f354287c/lib/Evaluator.py#L127
https://github.com/ultralytics/yolov5/blob/e808f2267d0164edb7bc45588c4fcda68c3dd8cb/utils/metrics.py#L64

mAP = torch.trapz(precision_curve, recall_curve)
